sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/JoinBenchmark.scala

```diff
@@ -212,18 +212,17 @@ class JoinBenchmark extends BenchmarkBase {
     sparkSession.conf.set("spark.sql.join.preferSortMergeJoin", "false")
     runBenchmark("shuffle hash join", N) {
       val df1 = sparkSession.range(N).selectExpr(s"id as k1")
-      val df2 = sparkSession.range(N / 5).selectExpr(s"id * 3 as k2")
+      val df2 = sparkSession.range(N / 3).selectExpr(s"id * 3 as k2")
```

Comment:
Thanks for fixing this although I'm afraid this might break again if we tweak the planner or the confs again. Can we please add an additional check that looks for `ShuffledHashJoinExec` node in the `SparkPlan`? Something along these lines:

```scala
assert(df.queryExecution.sparkPlan.find(_.isInstanceOf[ShuffledHashJoinExec]).isDefined)
```